src/Illuminate/Database/Eloquent/Builder.php

```diff
@@ -233,6 +233,8 @@ public function findOrNew($id, $columns = ['*'])
      */
     public function firstOrNew(array $attributes)
     {
+        $attributes = $this->model->newInstance()->fill($attributes)->attributesToArray();
```

Comment:
See the newest commit. Instead of forceFill() I just moved the instance line to the top of the method 
`$instance = $this->model->newInstance($attributes);` 
This was already in the method beforehand so it should not change the behavior of the method and still fix the bug that was reported.